src/node-plugin/lib/server.ts

```diff
@@ -114,19 +113,26 @@ export const serve = async (opts: ServeConfig) => {
   for (const key in pluginSet) {
     if (pluginSet.hasOwnProperty(key)) {
       const p = pluginSet[key];
-      p.grpcServer(server);
+      await p.grpcServer(server);
     }
   }
 
   opts.grpcHost = opts.grpcHost || '127.0.0.1';
   opts.grpcPort = opts.grpcPort || 0;
 
-  const address = `${opts.grpcHost}:${opts.grpcPort}`;
-  const boundPortNumber = await util.promisify(server.bindAsync).bind(server)(address, grpc.ServerCredentials.createInsecure());
-  if (boundPortNumber === 0) {
-    throw new Error(`failed to bind address=${address}, boundPortNumber=${boundPortNumber}`);
-  }
+  return new Promise<void>((resolve, reject) => {
```

Comment:
Why is `new Promise` preferred to `util.promisify`? I think the latter is cleaner and more modern since you avoid mixing `async` and callback style code.

Reply:
util.promisify didn't work with zeit/pkg that we're using to package plugin with node runtime.

Reply:
Ah, that's too bad :/